### ISSUE TITLE
UniversalProjectReader cannot read json

### ISSUE THREAD

**Issue by tesla-cat** [joniles/mpxj#399]
# code

- the json file is generated by `MpxjConvert`

```python
import jpype, mpxj

jpype.startJVM()

from net.sf.mpxj.reader import UniversalProjectReader

reader = UniversalProjectReader()
path = "temp.json"
project = reader.read(path)

for task in project.getTasks():
    print(task.getName())


```

# error

```cmd
Traceback (most recent call last):
  File "reader.py", line 11, in <module>
    for task in project.getTasks():
AttributeError: 'NoneType' object has no attribute 'getTasks'
```

**Comment by joniles** [comment]
Hello! You're receiving `NoneType` from `UniversalProjectReader` as it doesn't recognise the type of the file you've asked it to read. From the title of the issue do you mean that you are trying to read a JSON file? Is this the JSON structure which MPXJ writes? If so, this is currently only intended as a way for other applications to access the schedule data from MPXJ if they can't work with the Java/.Net classes directly, so there is no support for reading this format presently.

**Comment by tesla-cat** [comment]
as mentioned **the json file is generated by `MpxjConvert`**

suppose i want to build a web based Gantt editor, then I should be able to let people upload MS project XML files, and then let them download it back right?

So I think this requires `MSP XML ----> JSON` and then `JSON  -----> MSP XML` conversions right?  This seems to be the natural thing to do.

I mean MPXJ can already write to JSON, why shouldn't it be able to read from JSON?

**Comment by tesla-cat** [comment]
oh i see, thanks for your great effort on MPXJ!

but i thought reading JSON is much easier than XER / MPP?  

I have tried to write my own code to read MPP, but is in binary format without documentation on reading it.
also for XER, although it is in plain English and the structure is simple, but I have problems parsing the calendar data.

**Comment by tesla-cat** [comment]
by the way, i am not talking about any json file, but the json files generated by MPXJ itself

**Comment by joniles** [comment]
> by the way, i am not talking about any json file, but the json files generated by MPXJ itself

Yes, thats the one! It would be close to the same level of effort as one of the XML file format parsers.

> also for XER, although it is in plain English and the structure is simple, but I have problems parsing the calendar data.

You may find this code helpful to read the encoded calendar data if you wanted to continue writing your own XER parser:
https://github.com/joniles/mpxj/blob/master/src/main/java/net/sf/mpxj/primavera/StructuredTextParser.java
